src/index.rs

```diff
@@ -516,9 +563,41 @@ impl ResourceIndex {
         self.path2id.insert(path, entry);
     }
 
+    /// Removes the given resource ID from the index and returns an update
+    /// containing the deleted entries
+    pub fn forget_id(&mut self, old_id: ResourceId) -> Result<IndexUpdate> {
+        log::debug!("Forgetting a single entry in the index");
+
+        // Collect all paths associated with the old ID
+        let mut old_paths = Vec::new();
+        for (path, entry) in &self.path2id {
+            if entry.id == old_id {
+                old_paths.push(path.clone());
+            }
+        }
+
+        // Remove entries from path2id and id2path
+        for path in &old_paths {
+            self.path2id.remove(path);
+        }
+        self.id2path.remove(&old_id);
+
+        let mut deleted = HashSet::new();
+        deleted.insert(old_id);
+
+        Ok(IndexUpdate {
+            added: HashMap::new(),
+            deleted,
+        })
+    }
```

Comment:
From `index/update-one` branch:
```
    /// The caller must ensure that:
    /// * the index is up-to-date except this single id
    /// * the resource with this id has been indexed before
    /// * the resource with this id doesn't exist anymore
    /// 
    /// Should only be used if reactive updates are not possible.
    pub fn track_deletion(&mut self, id: ResourceId) -> Result<IndexUpdate> {
        log::debug!("Tracking a single deletion in the index");

        let indexed_path = self.id2path.get(&id);
        if indexed_path.is_none() {
            return Err(ArklibError::Path(
                "The id cannot be found in the index".into(),
            )); 
        }   

        let indexed_path = indexed_path.unwrap().clone();
        self.forget_entry(indexed_path.as_canonical_path(), id)?;

        Ok(IndexUpdate::default().deleted(id))
    }   
```

Reply:
The changes in forget_id in #72 don't work on their own because `ResourceIndex::forget_entry()` and `IndexUpdate` were also modified.